core/tests/protractor_desktop/additionalEditorFeaturesModals.js

```diff
@@ -380,6 +380,137 @@ describe('Full exploration editor', function() {
       await users.logout();
     });
 
+  it(
+    'should show toast message to inform when offline and online',
+    async function() {
+      await users.createUser('user16@editor.com', 'user16Editor');
+
+      await users.login('user16@editor.com');
+      await workflow.createExploration(true);
+      await explorationEditorPage.navigateToMainTab();
+
+      // Set network connection to offline.
+      await browser.driver.setNetworkConditions(
+        {
+          offline: true,
+          latency: 1000,
+          download_throughput: 0,
+          upload_throughput: 0});
+
+      // Check that toast message appeared when offline.
+      await waitFor.visibilityOf(
+        element(by.css('.protractor-test-toast-message')),
+        'Offline warning toast message taking too long to appear.');
+
+      // Set network connection to online.
+      await browser.driver.setNetworkConditions(
+        {
+          offline: false,
+          latency: 150,
+          download_throughput: 450 * 1024,
+          upload_throughput: 150 * 1024});
+
+      // Check that toast message appeared when reconnected.
+      await waitFor.visibilityOf(
+        element(by.css('.protractor-test-toast-message')),
+        'Online warning toast message taking too long to appear.');
+      await users.logout();
+    });
+
+  it(
+    'should be able to save changes when offline',
+    async function() {
+      await users.createUser('user17@editor.com', 'user17Editor');
+
+      await users.login('user17@editor.com');
+      await workflow.createExploration(true);
+      await explorationEditorPage.navigateToMainTab();
+
+      // Set network connection to offline.
+      await browser.driver.setNetworkConditions(
+        {
+          offline: true,
+          latency: 1000,
+          download_throughput: 0,
+          upload_throughput: 0});
+
```

Comment:
Can we also check for the visiblilty of offline toast message here?

Reply:
Done.